packages/datetime/src/dateRangePicker.tsx

```diff
@@ -495,6 +503,18 @@ export class DateRangePicker extends AbstractPureComponent<IDateRangePickerProps
         this.handleNextState(nextValue);
     };
 
+    private handleShortcutClick = (shortcut: IDateRangeShortcut) => {
+        const { dateRange, shouldChangeTime } = shortcut;
+        if (shouldChangeTime) {
+            const newDateRange: DateRange = [dateRange[0], dateRange[1]];
+            const newTimeRange: DateRange = [dateRange[0], dateRange[1]];
+            Utils.safeInvoke(this.props.onChange, newDateRange);
+            this.setState({ value: newDateRange, time: newTimeRange });
```

Comment:
Haven't looked at this component in a while, but just to clarify: do we really not need to invoke `getStateChange`? What if:

(1) The current date and time is January 1st at 8:00am, and
(2) I'm looking at the January and February in the Date Range Input, and
(3) I select a shortcut for "Last 12 hours".

We'd expect the view to shift to reveal the start date of the range (i.e., December 31 of the prior year for the above example). But I don't think the calendar month would change with this code.

Reply:
You are right. I've added a call to `getStateChange` to fix this.